backend/test_real_chinese_website.py

```
@@ -0,0 +1,249 @@
+                mock_response._content = case['content'].encode('utf-8')
+
+            mock_response.headers = {'content-type': f'text/html; charset={case["encoding"] or ""}'}
+
+            # 模拟text属性的行为
+            if case['encoding'] == 'gbk':
+                mock_response.text = case['content']  # 假设能正确解码
+            elif case['encoding'] is None:
+                # 模拟requests的默认行为（可能使用latin-1）
+                try:
+                    mock_response.text = case['content'].encode('utf-8').decode('latin-1')
+                except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)